### PR TITLE
Add archive configs

### LINKED ISSUE #2150: Add Configs to set Archive windows
* **What are we trying to do?**
Make the Archive timelines for Dropoff and Fulfilled patients a fund-controlled configuration

Mockup:
archive config mockup

* **What feature or behavior is this required for?**
Some funds have frustrations around our data anonymizing too early for their processes. We don't want to set everyone to a longer timeline, as that increased issues around folks who need our support again within that window due to the phone number being a unique ID. Instead, we want to allow funds to make that trade off decision themselves.

* **How could we solve this issue? (Not knowing is okay!)**
Add a new config field in the [Config model](https://github.com/DCAFEngineering/dcaf_case_management/blob/main/app/models/config.rb), set reasonable defaults and bounds on the input, and use that value instead of the literals in the [Patient model](https://github.com/DCAFEngineering/dcaf_case_management/blob/851bf72f0b67898bce06fceb32829d28ec1af346/app/models/patient.rb#L260-L271)

* **Anything else?**
See #1914 for the full discussion. CC @verbingthenoun

### PR DESCRIPTION
**Pull request by elimbaum**
I rule and have completed some work on Case Manager that's ready for review!

Adds config for custom patient-archive timelines.

This pull request makes the following changes:
* Adds new configs with bounds-checking
* Replaces constants in patient model with those custom values

Screen Shot 2021-05-09 at 17 50 51

*new configs*

It relates to the following issue #s: 
* Fixes #2150 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Comment by elimbaum** [comment]
Wording is now "Days to keep fulfilled patients" and "Days to keep all patients".

**Comment by elimbaum** [comment]
Requesting review pending decision on min/max/default values!

**Review by colinxfleming — APPROVED** [review]
This is good stuff - very well constructed PR here. I have a few suggestions on screws to tighten but this looks good to me generally.

I'd like to get @verbingthenoun to weigh in here to confirm this will get their team out of purgatory -- this will make it so that patients can get not-archived up to 1.5 years after initial call. It still hinges archive timelines on the initial call, so I think you still should not click the audited button until you have verified that someone is Definitely Absolutely Good to Go and you don't need their data anymore. (It might be easier to do this on a call, lmk if that's the case?)

I'm clicking approve from a code perspective though. Nice work.

File: test/models/config_test.rb [hunk 1]
```diff
@@ -121,6 +121,36 @@ class ConfigTest < ActiveSupport::TestCase
...
+        
+        # in bounds
+        c.config_value = { options: ["120"] }
+        assert c.valid?
+        
+        # high edge
+        c.config_value = { options: ["550"] }
+        assert c.valid?
+        
+        c.config_value = { options: ["1000"] }
```
[the remark this answers is not in the document]
Comment (elimbaum) [reply test/models/config_test.rb]: good pt. no reason to check insane values if sane values will do.

[the remark this answers is not in the document]
Comment (verbingthenoun) [reply test/models/config_test.rb]: This sounds like it will work from my end. Thank you!

**Comment by colinxfleming** [comment]
Planning on merging this over the weekend since we've gotten the okay from @verbingthenoun

**Pull request merged**